runtime/testy/runtime.py

```diff
@@ -42,31 +45,37 @@ def runtime():
     spawnProcess(PROCESS_NAMES.UDP_RECEIVE_PROCESS, startUDPReceiver)
     spawnProcess(PROCESS_NAMES.HIBIKE, startHibike)
     while True:
-      if restartCount >= 5 or emergency_stopped:
-        print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
-        print("Too many restarts, terminating")
+      if restartCount >= 3 or emergency_stopped:
+        if not testMode:
+          print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
+          print("Too many restarts, terminating")
         break
-      print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
-      print("Starting studentCode attempt: %s" % (restartCount,))
-      spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode)
+      if not testMode:
+        print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
+        print("Starting studentCode attempt: %s" % (restartCount,))
+      spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode, testName, maxIter)
       while True:
         newBadThing = badThingsQueue.get(block=True)
         print(newBadThing.event)
-        print(newBadThing.data)
+        if not testMode:
+          print(newBadThing.data)
         if newBadThing.event in restartEvents:
-          if(not emergency_stopped and newBadThing.event is BAD_EVENTS.EMERGENCY_STOP):
+          if (not emergency_stopped and newBadThing.event is BAD_EVENTS.EMERGENCY_STOP):
             emergency_stopped = True #somehow kill student code using other method? right now just restarting on e-stop
           break
       stateQueue.put([SM_COMMANDS.RESET, []])
-      terminate_process(allProcesses[PROCESS_NAMES.STUDENT_CODE])
+      terminate_process(PROCESS_NAMES.STUDENT_CODE)
       restartCount += 1
-    print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
+    if not testMode:
+      print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
     print("Funtime Runtime is done having fun.")
     print("TERMINATING")
-  except:
+  except Exception as e:
     print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
     print("Funtime Runtime Had Too Much Fun")
-    print(traceback.print_exception(*sys.exc_info()))
+    print(e)
```

Comment:
I trust that you know what you're doing with regards to changing the exception printing code. (TBH I don't know what the difference between the old and the new code on this line is)